Sources/SotoCore/Credential/ConfigFileCredentialProvider.swift

```diff
@@ -12,144 +12,102 @@
-            }
-    }
-
-    static func sharedCredentials(from byteBuffer: ByteBuffer, for profile: String) throws -> StaticCredential {
-        let string = byteBuffer.getString(at: 0, length: byteBuffer.readableBytes)!
-        var parser: INIParser
-        do {
-            parser = try INIParser(string)
-        } catch INIParser.Error.invalidSyntax {
-            throw ConfigFileError.invalidCredentialFileSyntax
-        }
-
-        guard let config = parser.sections[profile] else {
-            throw ConfigFileError.missingProfile(profile)
+        return ConfigFileLoader.loadSharedCredentials(
+            credentialsFilePath: credentialsFilePath,
+            configFilePath: configFilePath,
+            profile: profile,
+            context: context
+        )
+        .flatMapThrowing { credentials, config in
+            return try credentialProvider(from: credentials, config: config, context: context)
         }
+    }
 
-        guard let accessKeyId = config["aws_access_key_id"] else {
-            throw ConfigFileError.missingAccessKeyId
+    /// Generate credential provider based on shared credentials and profile configuration
+    ///
+    /// Credentials file settings have precedence over profile configuration settings
+    /// https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html#cli-configure-quickstart-precedence
+    ///
+    /// - Parameters:
+    ///   - credentialsBuffer: contents of AWS shared credentials file (usually `~/.aws/credentials`
+    ///   - configByteBuffer: contents of AWS profile configuration file (usually `~/.aws/config`
+    ///   - context: credential provider factory context
+    /// - Returns: Credential Provider (StaticCredentials or STSAssumeRole)
+    static func credentialProvider(
+        from credentials: ConfigFileLoader.ProfileCredentials,
+        config: ConfigFileLoader.ProfileConfig?,
+        context: CredentialProviderFactory.Context
+    ) throws -> CredentialProvider {
+        let staticCredential = StaticCredential(
+            accessKeyId: credentials.accessKey,
+            secretAccessKey: credentials.secretAccessKey,
+            sessionToken: credentials.sessionToken
+        )
+
+        // If `role_arn` and `sourcer_profile` are defined, temporary credentials must be loaded via STS Assume Role operation
+        if let roleArn = credentials.roleArn ?? config?.roleArn,
+           let _ = credentials.sourceProfile ?? config?.sourceProfile
+        {
+            // Proceed with STSAssumeRole operation
+            let sessionName = credentials.roleSessionName ?? config?.roleSessionName ?? UUID().uuidString
+            let request = STSAssumeRoleRequest(roleArn: roleArn, roleSessionName: sessionName)
+            let region = config?.region ?? .useast1
+            let provider = STSAssumeRoleCredentialProvider(request: request, credentialProvider: .default, region: region, httpClient: context.httpClient)
```

Comment:
Ah, right, forgot about updating this. Thanks!

Reply:
@adam-fowler, looks like `STSAssumeRoleCredentialProvider` takes a `CredentialProviderFactory`, instead of a `CredentialProvider`. How do I make a provider factory from `StaticCredential`?

Reply:
.static(accessKeyId: ...)

Reply:
I see, thanks